sparse/unit_test/Test_Sparse_sptrsv.hpp

```diff
@@ -823,25 +592,21 @@ template <typename scalar_t, typename lno_t, typename size_type,
           typename device>
 void test_sptrsv_streams() {
   using TestStruct = Test::SptrsvTest<scalar_t, lno_t, size_type, device>;
-
-  TestStruct::run_test_sptrsv_streams(0, 1);
-  TestStruct::run_test_sptrsv_streams(0, 2);
-  TestStruct::run_test_sptrsv_streams(0, 3);
-  TestStruct::run_test_sptrsv_streams(0, 4);
-  TestStruct::run_test_sptrsv_streams(1, 1);
-  TestStruct::run_test_sptrsv_streams(1, 2);
-  TestStruct::run_test_sptrsv_streams(1, 3);
-  TestStruct::run_test_sptrsv_streams(1, 4);
-
+  std::vector<SPTRSVAlgorithm> algs = {SPTRSVAlgorithm::SEQLVLSCHD_RP,
+                                       SPTRSVAlgorithm::SEQLVLSCHD_TP1};
 #if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE)
   if (std::is_same<lno_t, int>::value &&
       std::is_same<typename device::execution_space, Kokkos::Cuda>::value) {
-    TestStruct::run_test_sptrsv_streams(2, 1);
-    TestStruct::run_test_sptrsv_streams(2, 2);
-    TestStruct::run_test_sptrsv_streams(2, 3);
-    TestStruct::run_test_sptrsv_streams(2, 4);
+    algs.push_back(SPTRSVAlgorithm::SPTRSV_CUSPARSE);
```

Comment:
Also need to check ``size_type`` is int?

Reply:
It looks like I did preserve the semantics of the original code:

```
#if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE)
  if (std::is_same<lno_t, int>::value &&
      std::is_same<typename device::execution_space, Kokkos::Cuda>::value) {
    TestStruct::run_test_sptrsv_streams(2, 1);
    TestStruct::run_test_sptrsv_streams(2, 2);
    TestStruct::run_test_sptrsv_streams(2, 3);
    TestStruct::run_test_sptrsv_streams(2, 4);
  }
#endif
```

I don't know why size_type was not being checked here but it is earlier in the file. Do you have any thoughts on which one is right? It sounds like you think we should always be checking size_type==int.

Reply:
Yes, the other one was correct. Unless something has changed in cusparse recently, it doesn't support having offsets and entries of different types. They have to be both 32 bit, or both 64 bit.